### DISCUSSION TITLE
Choosing mail recipient based on alert properties

### DISCUSSION THREAD

**Discussion by tflorac** [jertel/elastalert2#632] [Q&A] [answered]
Hi,
I have created ES alerts based on heartbeat which is polling several services.
In ElastAlert, a rule is checking for these alerts, and is sending an email as soon as a service is unavailable.
My question is: is it possible to choose email recipients based on properties of the alert which was triggered, without creating several ElastAlert rules?
Best regards,
Thierry

**Comment by nsano-rururu** [comment]
I don't think there is such a function

**Comment by jertel** [comment, marked as answer]
There are two options:

1. Use the `email_from_field` to email an address or username embedded in the alert data. See [docs](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#email).
2. You can create an small [enhancement](https://elastalert2.readthedocs.io/en/latest/recipes/adding_enhancements.html#enhancements) which looks at the match data and sets the `email` field to the desired recipient(s).

**Reply by tflorac** [reply]
Very cool!
Additional questions anyway:
 - if using an `email_from_field` option, can the matching attribute contain several addresses?
 - is the `email` option used **only** when `email_from_field` is not set; this is, if I want a recipient to always be notified, do I have to define his address in alert data and email option?

**Reply by nsano-rururu** [reply]
About email_from_field
https://github.com/Yelp/elastalert/issues/1064
https://github.com/Yelp/elastalert/pull/1163
document
https://elastalert2.readthedocs.io/en/latest/ruletypes.html#email
>email_from_field: Use a field from the document that triggered the alert as the recipient. If the field cannot be found, the email value will be used as a default. Note that this field will not be available in every rule type, for example, if you have use_count_query or if it’s type: flatline. You can optionally add a domain suffix to the field to generate the address using email_add_domain. It can be a single recipient or list of recipients. For example, with the following settings:
